src/Compilers/CSharp/Portable/BoundTree/BoundTreeWalker.cs

```diff
@@ -129,9 +127,9 @@ protected BoundTreeWalkerWithStackGuardWithoutRecursionOnTheLeftOfBinaryOperator
 
             this.Visit(current);
 
-            while (rightOperands.Count > 0)
+            while (rightOperands.TryPop(out current))
```

Comment:
>while (rightOperands.TryPop(out current)) [](http://example.com/codeflow?start=12&length=41)

Since you decided to change this code, consider using a loop with a post-condition